plotly/tools.py

```diff
@@ -2871,3 +2934,188 @@ def get_candle_decrease(self):
         return (decrease_x, decrease_close, decrease_dif,
                 stick_decrease_y, stick_decrease_x)
 
+class _Dendrogram(FigureFactory):
+    
+    """
+    Refer to FigureFactory.create_dendrogram() for docstring.
+    """
+
+    def __init__(self, X, orientation='bottom', labels=None, colorscale=None, \
+                 width="100%", height="100%", xaxis='xaxis', yaxis='yaxis' ):
+        self.orientation = orientation
+        self.labels = labels
+        self.xaxis = xaxis
+        self.yaxis = yaxis
+        self.data = []
+        self.leaves = []
+        self.sign = {self.xaxis: 1, self.yaxis: 1}
+        self.layout = {self.xaxis: {}, self.yaxis: {}}
```

Comment:
Hm, maybe I'm unclear on which xaxis/yaxis you're trying to store here. There's the *axis reference* which is when you say something like `'xaxis': 'x1'` and then there's the *layout axis object* which is when you say something like `'xaxis2': {'domain': [0, 1]}`.

So, I *think* you're making them serve two purposes, using the full string to name the layout object and trying to take the last digit and use it as a *reference*. I'd suggest you write a bit of logic in there that handles the case when the last character *isn't* a digit:

```python
try:
    axis_index = int(axis_string[-1])
except ValueError:
    axis_index = ''
else:
    axis_char = axis_string[0]
    axis_reference = '{}{}'.format(axis_char, axis_index)
```

Again, sorry if I'm misinterpreting. I think just clearing up the test suite will sort this out.